sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/AfterSynchronizedProcessingTime.java

```diff
@@ -18,21 +18,22 @@
 package org.apache.beam.sdk.transforms.windowing;
 
 import com.google.common.base.Objects;
-import java.util.Collections;
 import java.util.List;
-import org.apache.beam.sdk.transforms.SerializableFunction;
-import org.apache.beam.sdk.util.TimeDomain;
+import org.apache.beam.sdk.transforms.windowing.Trigger.OnceTrigger;
 import org.joda.time.Instant;
 
 /**
- * A trigger that fires after synchronized processing time has reached a shared
- * threshold between upstream workers.
+ * A trigger that fires after synchronized processing time has reached the processing time of the
+ * first element's arrival.
+ *
+ * <p>This is for internal, primarily as a "continuation trigger" for {@link AfterProcessingTime}
+ * triggers. In that use, this trigger is ready as soon as all upstream workers processing time
+ * clocks have caught up to the moment that input arrived.
  */
-public class AfterSynchronizedProcessingTime extends AfterDelayFromFirstElement {
+public class AfterSynchronizedProcessingTime extends OnceTrigger {
 
   public AfterSynchronizedProcessingTime() {
```

Comment:
Done. Been that way for a while, but hey as usual a static factory method clarifies things.